src/components/carousel/index.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
+import {
+  CarouselWrapper,
+  CarouselImg,
+  LeftBtn,
+  RightBtn,
+  ProgressBarWrapper,
+  ProgressBar,
+} from './index.style';
+
+interface CarouselProps {
+  images: string[]; // 이미지 배열
+  width?: number; // 슬라이드 하나의 너비
+  height?: number; // 슬라이드 높이
+  showButtons?: boolean; // 버튼 표시 여부
+  autoPlay?: boolean; // 자동 재생 여부
+  autoPlayInterval?: number; // 자동 재생 시간 간격
+}
+
+const Carousel: React.FC<CarouselProps> = ({
+  images,
+  width = 480,
+  height = 408,
+  showButtons = true,
+  autoPlay = false,
+  autoPlayInterval = 3000,
+}) => {
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const handlePrev = () => {
+    setCurrentIndex((prevIndex) => (prevIndex === 0 ? images.length - 1 : prevIndex - 1));
+  };
+  const handleNext = () => {
+    setCurrentIndex((prevIndex) => (prevIndex === images.length - 1 ? 0 : prevIndex + 1));
+  };
+
+  // 자동 재생 기능
+  React.useEffect(() => {
```

Comment:
`useState`처럼 `useEffect`를 import해서 사용해 주세요!
```suggestion
useEffect(() => {
```